solver/solver.go

```diff
@@ -59,6 +62,12 @@ func Solve(doneCh chan struct{}, syncer *diff.Syncer,
 			// fire the request to Kong
 			result, err = r.Do(e.Kind, e.Op, e)
 			if err != nil {
+				// Make only 500 errors retryable - this is what Kong returns when e.g. the
+				// database is temporarily overloaded.
+				// TODO: Replace with better check when/if go-kong exposes status code
+				if !strings.HasPrefix(err.Error(), "500") {
```

Comment:
- Instead of parsing the error message, please update go-kong to `v0.14.0` and extract `Code()` from the [`APIError` type](https://github.com/Kong/go-kong/blob/f71c83b51f21bf129b63a60aa88447203d6bfb8b/kong/error.go#L8).
- Then, instead of literal `500`, use `http.StatusInternalServerError`

Reply:
Glad to see the addition to go-kong, done.